build_tools/third_party/llvm-project/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+###############################################################################
+# Re-export LLVM header-only targets as header-only libraries.                #
+###############################################################################
+
+add_library(MLIRLinalgStructuredOpsIncGenLib INTERFACE)
+add_dependencies(MLIRLinalgStructuredOpsIncGenLib
+  MLIRLinalgStructuredOpsIncGen
+)
```

Comment:
This approach seems reasonable enough to me at this scale. If we end up with 10s of these across the project then we'd want to be more careful.

I'm a bit foggy on the specifics right now, but here's another general idea to consider: we already have [`iree_compiler_cc_library`](https://github.com/google/iree/blob/18b9df4ac41f7cbf86ec0e036771b39f9f3420f4/iree/build_defs.oss.bzl#L49-L60) in Bazel [getting converted](https://github.com/google/iree/blob/18b9df4ac41f7cbf86ec0e036771b39f9f3420f4/build_tools/bazel_to_cmake/bazel_to_cmake_converter.py#L335-L336) to [`iree_cc_library`](https://github.com/google/iree/blob/main/build_tools/cmake/iree_cc_library.cmake) in CMake. We could add another argument to `iree_compiler_cc_library` (Bazel) that would get converted to a new argument on `iree_cc_library` (CMake) that would use `add_dependences`.

Might want Geoffrey to weigh in when he gets back :)

Reply:
I didn't know about `iree_compiler_cc_library`. Looks like it's new. Your idea seems good Scott, but the approach in this PR also seems like an improvement. How about we merge this and then try out the `iree_compiler_cc_library` idea?

Reply:
Filed #9146 to track this.